aiokafka/fetcher.py

```diff
@@ -0,0 +1,579 @@
+                contains OffsetOutOfRangeError and the default_reset_policy is
+                None
+        """
+        if not self._offset_out_of_range_partitions:
+            return
+
+        current_out_of_range_partitions = {}
+
+        # filter only the fetchable partitions
+        for partition, offset in self._offset_out_of_range_partitions:
+            if not self._subscriptions.is_fetchable(partition):
+                log.debug("Ignoring fetched records for %s since it is no"
+                          " longer fetchable", partition)
+                continue
+            position = self._subscriptions.assignment[partition].position
+            # ignore partition if the current position != offset
+            # in FetchResponse e.g. after seek()
+            if position is not None and offset == position:
+                current_out_of_range_partitions[partition] = position
+
+        self._offset_out_of_range_partitions.clear()
+        if current_out_of_range_partitions:
+            raise Errors.OffsetOutOfRangeError(
+                current_out_of_range_partitions)
+
+    def _raise_if_unauthorized_topics(self):
+        """Check FetchResponses for topic authorization failures.
+
+        Raises:
+            TopicAuthorizationFailedError
+        """
+        if self._unauthorized_topics:
+            topics = set(self._unauthorized_topics)
+            self._unauthorized_topics.clear()
+            raise Errors.TopicAuthorizationFailedError(topics)
+
+    def _raise_if_record_too_large(self):
+        """Check FetchResponses for messages larger than the max per partition
+
+        Raises:
+            RecordTooLargeError: if there is a message larger than fetch size
+        """
+        if not self._record_too_large_partitions:
+            return
+
+        copied_record_too_large_partitions = dict(
+            self._record_too_large_partitions)
+        self._record_too_large_partitions.clear()
+
+        raise RecordTooLargeError(
+            "There are some messages at [Partition=Offset]: %s "
+            " whose size is larger than the fetch size %s"
+            " and hence cannot be ever returned."
+            " Increase the fetch size, or decrease the maximum message"
+            " size the broker will allow.",
+            copied_record_too_large_partitions,
+            self._max_partition_fetch_bytes)
+
+    def _unpack_message_set(self, tp, messages):
+        for offset, size, msg in messages:
+            if self._check_crcs and not msg.validate_crc():
+                raise Errors.InvalidMessageError(msg)
+            elif msg.is_compressed():
+                for record in self._unpack_message_set(tp, msg.decompress()):
+                    yield record
+            else:
+                key, value = self._deserialize(msg)
+                yield ConsumerRecord(
+                    tp.topic, tp.partition, offset, key, value)
+
+    def _message_generator(self, partitions):
```

Comment:
Pls remove this generator. It's adding complexity, which is most certainlty not needed here. Just add a function, that will post process a single message.